src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -184,3 +187,108 @@ def print_revert_report(self, *, delete_managed: bool) -> bool | None:
+            self._ws.schemas.create(name=to_schema, catalog_name=to_catalog)
+
+        tables = self._ws.tables.list(catalog_name=from_catalog, schema_name=from_schema)
+        table_tasks = []
+        view_tasks = []
+        filtered_tables = [table for table in tables if from_table in [table.name, "*"]]
+        for table in filtered_tables:
+            try:
+                self._ws.tables.get(full_name=f"{to_catalog}.{to_schema}.{table.name}")
+                logger.warning(
+                    f"table {from_table} already present in {from_catalog}.{from_schema}. skipping this table..."
+                )
+                continue
+            except NotFound:
+                if table.table_type and table.table_type.value in ("EXTERNAL", "MANAGED"):
+                    table_tasks.append(
+                        partial(self._migrate_uc_table, from_catalog, from_schema, table.name, to_catalog, to_schema)
+                    )
+                else:
+                    view_tasks.append(
+                        partial(
+                            self._migrate_uc_view,
+                            from_catalog,
+                            from_schema,
+                            table.name,
+                            to_catalog,
+                            to_schema,
+                            table.view_definition,
+                        )
+                    )
+        Threads.strict(name="creating tables", tasks=table_tasks)
+        logger.info(f"migrated {len(list(table_tasks))} tables to the new schema {to_schema}.")
+        Threads.strict(name="creating views", tasks=view_tasks)
+        logger.info(f"migrated {len(list(view_tasks))} views to the new schema {to_schema}.")
+
+    def _migrate_uc_table(
+        self,
+        from_catalog: str,
+        from_schema: str,
+        from_table: str,
+        to_catalog: str,
+        to_schema: str,
+    ) -> bool:
+        from_table_name = f"{from_catalog}.{from_schema}.{from_table}"
+        to_table_name = f"{to_catalog}.{to_schema}.{from_table}"
+        try:
+            create_sql = str(next(self._backend.fetch(f"SHOW CREATE TABLE {from_table_name}"))[0])
+            create_sql = create_sql.replace(from_catalog, to_catalog).replace(from_schema, to_schema)
+            logger.debug(f"Creating table {from_table_name}.")
+            self._backend.execute(create_sql)
+            grants = self._ws.grants.get(securable_type=SecurableType.TABLE, full_name=from_table_name)
+            if grants.privilege_assignments is None:
+                return True
+            grants_changes = [
+                PermissionsChange(add=pair.privileges, principal=pair.principal)
+                for pair in grants.privilege_assignments
+            ]
+            self._ws.grants.update(securable_type=SecurableType.TABLE, full_name=to_table_name, changes=grants_changes)
+            return True
+        except DatabricksError:
```

Comment:
This error is too generic and hides the real problems with the code. You are allowed to catch only subclasses of it. The other errors will bubble up